commit 87835f30e4f72dbfcde253eb8d1c2353c05348fb
Author: Dmitry Bogdanov <dmitry.bogdanov@aquantia.com>
Date:   Wed Aug 8 14:06:32 2018 +0300

    net: aquantia: Fix IFF_ALLMULTI flag functionality
    
    [ Upstream commit 11ba961c916127651e12af6cad3891f8aeb25aa9 ]
    
    It was noticed that NIC always pass all multicast traffic to the host
    regardless of IFF_ALLMULTI flag on the interface.
    The rule in MC Filter Table in NIC, that is configured to accept any
    multicast packets, is turning on if IFF_MULTICAST flag is set on the
    interface. It leads to passing all multicast traffic to the host.
    This fix changes the condition to turn on that rule by checking
    IFF_ALLMULTI flag as it should.
    
    Fixes: b21f502f84be ("net:ethernet:aquantia: Fix for multicast filter handling.")
    Signed-off-by: Dmitry Bogdanov <dmitry.bogdanov@aquantia.com>
    Signed-off-by: David S. Miller <davem@davemloft.net>
    Signed-off-by: Greg Kroah-Hartman <gregkh@linuxfoundation.org>

diff --git a/drivers/net/ethernet/aquantia/atlantic/hw_atl/hw_atl_b0.c b/drivers/net/ethernet/aquantia/atlantic/hw_atl/hw_atl_b0.c
index 956860a69797..3bdab972420b 100644
--- a/drivers/net/ethernet/aquantia/atlantic/hw_atl/hw_atl_b0.c
+++ b/drivers/net/ethernet/aquantia/atlantic/hw_atl/hw_atl_b0.c
@@ -762,7 +762,7 @@ static int hw_atl_b0_hw_packet_filter_set(struct aq_hw_s *self,
 
 	hw_atl_rpfl2promiscuous_mode_en_set(self, IS_FILTER_ENABLED(IFF_PROMISC));
 	hw_atl_rpfl2multicast_flr_en_set(self,
-					 IS_FILTER_ENABLED(IFF_MULTICAST), 0);
+					 IS_FILTER_ENABLED(IFF_ALLMULTI), 0);
 
 	hw_atl_rpfl2_accept_all_mc_packets_set(self,
 					       IS_FILTER_ENABLED(IFF_ALLMULTI));
diff -uN a/1.txt b/1.txt
--- a/dummy/rpi_1008_87835f30e4f72dbfcde253eb8d1c2353c05348fb.txt	1970-01-01 00:00:00.000000000 +0000
+++ b/dummy/rpi_1008_87835f30e4f72dbfcde253eb8d1c2353c05348fb.txt	2013-12-23 04:07:40.000000000 +0000
@@ -0,0 +1 @@
+dummy file to ensure patch has content.
